styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -309,6 +318,53 @@ private void processInstance(StyxConfig config, Map<String, Resource> resources,
     sendDequeue(instance, runState, instanceResourceRefs);
   }
 
+  /**
+   * Shuffle instances. For the same workflow ID, older instances will appear early in the list.
+   */
+  @VisibleForTesting
+  List<WorkflowInstance> shuffleInstances(Set<WorkflowInstance> activeInstances) {
+    var groups = activeInstances
+        .stream()
+        .collect(groupingBy(WorkflowInstance::workflowId, LinkedHashMap::new,
```

Comment:
Using `LinkedHashMap` to have predictable iteration to simplify test

Reply:
I guess this comment could live with the code because it would be hard to recall why order matters.